java/vector/src/main/java/org/apache/arrow/vector/BaseNullableVariableWidthVector.java

```diff
@@ -761,4 +741,57 @@ protected final void handleSafe(int index, int dataLength) {
          reallocValueBuffer();
       }
    }
+
+
+   /******************************************************************
+    *                                                                *
+    *                helper methods currently                        *
+    *                used by JsonFileReader and                      *
+    *                JsonFileWriter                                  *
+    *                                                                *
+    ******************************************************************/
+
+
+   public static ArrowBuf setSafeJsonHelper(ArrowBuf data, ArrowBuf offset,
+                                            BufferAllocator allocator, int index, byte[] value,
+                                            int valueCount) {
+      if (data == null) {
+         data = allocator.buffer(INITIAL_BYTE_COUNT);
+      }
+      final int currentBufferCapacity = data.capacity();
+      final int currentStartOffset = offset.getInt(index * OFFSET_WIDTH);
+      while (currentBufferCapacity < currentStartOffset + value.length) {
+         final ArrowBuf newBuf = allocator.buffer(currentBufferCapacity * 2);
+         newBuf.setBytes(0, data, 0, currentBufferCapacity);
+         data.release();
+         data = newBuf;
+      }
+      data.setBytes(currentStartOffset, value, 0, value.length);
+      if (index == (valueCount - 1)) {
+         data.writerIndex(offset.getInt(valueCount * OFFSET_WIDTH));
+      }
+      return data;
+   }
+
+   public static byte[] get(final ArrowBuf data, final ArrowBuf offset, int index) {
```

Comment:
Same as above.

Reply:
Done.

Reply:
Sorry, I really don't like having these functions as a part of the vector API just for the JSON reader. I am not sure what's the best way to do this though - Can you please explain why do we need these methods? The original JSON reader/writer use these - 
https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java#L344

https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileWriter.java#L240

Reply:
This is discussed https://github.com/apache/arrow/pull/1203#issuecomment-338690930